testing/unit_tests/tinitTests.m

```diff
@@ -61,4 +61,507 @@ function testparsexpTask1List(testCase)
+
+    %make R7 and R10 spontaneous (also same as in T0001)
+    prepDataTest1 = prepINITModel(testModel, {}, {'R7';'R10'}, false, {}, 's');
+    verifyTrue(testCase, all(strcmp(prepDataTest1.refModel.rxns(prepDataTest1.toIgnoreExch | prepDataTest1.toIgnoreSpont), {'R1';'R7';'R8';'R10'})))
+    tst1ResModel1 = ftINIT(prepDataTest1,arrayData1.tissues{1},[],[],arrayData1,[],getINITSteps(),true,true,testParams);
+    %the model should now change to include the "correct" path (including 'R2') and 
+    %skip R9/R10:
+    verifyTrue(testCase, all(strcmp(tst1ResModel1.rxns,{'R1';'R2';'R4';'R6';'R7';'R8'})))
+    %now, test to add the metabolite f - this should turn back the favor to R9/R10:
+    tst1ResModel1 = ftINIT(prepDataTest1,arrayData1.tissues{1},[],[],arrayData1,{'f'},getINITSteps(),true,true,testParams);
+    %R9 should now be included, the presence of R2 is random
+    if length(tst1ResModel1.rxns) == 7
+        verifyTrue(testCase, all(strcmp(tst1ResModel1.rxns,{'R1';'R4';'R6';'R7';'R8';'R9';'R10'})))
+    else
+        verifyTrue(testCase, all(strcmp(tst1ResModel1.rxns,{'R1';'R2';'R4';'R6';'R7';'R8';'R9';'R10'})))
+    end
+    %now, test to add the metabolite a, e, f - this should give the same result:
+    tst1ResModel1 = ftINIT(prepDataTest1,arrayData1.tissues{1},[],[],arrayData1,{'f';'a';'e'},getINITSteps(),true,true,testParams);
+    %Should be the same as above (R2 is random)
+    if length(tst1ResModel1.rxns) == 7
+        verifyTrue(testCase, all(strcmp(tst1ResModel1.rxns,{'R1';'R4';'R6';'R7';'R8';'R9';'R10'})))
+    else
+        verifyTrue(testCase, all(strcmp(tst1ResModel1.rxns,{'R1';'R2';'R4';'R6';'R7';'R8';'R9';'R10'})))
+    end
+    
+    %now, test to add the metabolite b - this should turn on R2 and R3/R5 and turn off R9:
+    tst1ResModel1 = ftINIT(prepDataTest1,arrayData1.tissues{1},[],[],arrayData1,{'b'},getINITSteps(),true,true,testParams);
+    verifyTrue(testCase, all(strcmp(tst1ResModel1.rxns,{'R1';'R2';'R3';'R4';'R5';'R6';'R7';'R8'})))
+
+    %now on model 5 to test reactions that are not linearly merged (testModel has only merged rxns)
+    testModel5 = getTstModel5();
+    arrayData1.genes = testModel5.genes;
+    arrayData1.tissues = {'a'};
+    arrayData1.levels = getExprForRxnScore(getTstModel5RxnScores());
+    arrayData1.threshold = 1;
+    
+    prepDataTest5 = prepINITModel(testModel5, {}, {'R7';'R10'}, false, {}, 's');
+    tst1ResModel1 = ftINIT(prepDataTest5,arrayData1.tissues{1},[],[],arrayData1,{},getINITSteps(),true,true,testParams);
+    %We expect the 'true' path, i.e. through R2, not R9/R10 or R11-R14
+    verifyTrue(testCase, all(strcmp(tst1ResModel1.rxns,{'R1';'R2';'R4';'R6';'R7';'R8'})), 1)
+
+    %now add metabolite g
+    tst1ResModel1 = ftINIT(prepDataTest5,arrayData1.tissues{1},[],[],arrayData1,{'g'},getINITSteps(),true,true,testParams);
+    %We expect R2 to be replaced with R11 and R13
+    verifyTrue(testCase, all(strcmp(tst1ResModel1.rxns,{'R1';'R4';'R6';'R7';'R8';'R11';'R13'})), 1)
```

Comment:
Here R2 is consistently not replaced. Running with Gurobi.

Reply:
Hmm, I will have to take a look at this. All passed when I did the pull request. Perhaps there is a mistake in this test case, or in the code.

Reply:
It's the only (sub-)test that fails, the others pass fine.

Reply:
I have pushed some fixes for this now, the test case was simply not right.

Reply:
Now it passes!